test/linter.js

```diff
@@ -4,23 +4,124 @@
 'use strict';
 var
   execSync = require('child_process').execSync,
-  utils = require('./util/utils');
+  utils = require('./util/utils'),
+  fs = require('fs'),
+  literate = require('@microsoft.azure/literate'),
+  async_io_1 = require("@microsoft.azure/async-io"),
+  path = require('path');
+
+async function readConfigFile(file, tag) {
+  // autorest configuration type
+  const msgs = new literate.MessageEmitter();
+  msgs.Message.Subscribe((src, each) => console.log(each.Text));
+  const cfg = new literate.Configuration("\n> see https://aka.ms/autorest", new literate.DiskFileSystem("readme.md"), async_io_1.ResolveUri(async_io_1.CreateFolderUri(process.cwd()), async_io_1.CreateFileUri(path.resolve(file)) || "."), {
+    "input-file": [],
+  });
+  return await cfg.CreateView(msgs, true, { tag: tag });
+}
+
+async function getTagsMapFromConfig(args) {
+  if (!fs.existsSync(path.resolve(args.file))) {
+    console.error('config file invalid. cannot read tags from file ' + config);
+    return null;
+  }
+
+  const allTags = scanForTags(fs.readFileSync(args.file));
+  const result = {};
+  for (const each of allTags) {
+    result[each] = (await readConfigFile(args.file, each))["input-file"];
+  }
+  return result;
+}
+
+function scanForTags(content) {
+  const result = new Array();
+  const rx = /\$\(tag\)(.*)/g;
+  let match = rx.exec(content);
+  while (match) {
+    const vrx = /['"](.*?)['"]/g;
+    let v = vrx.exec(match[1]);
+    if (v && v.length && result.indexOf(v[1]) == -1) {
+      result.push(v[1]);
+    }
+    match = rx.exec(content);
+  }
+  return result;
+}
+
+async function getTagsFromConfig(config) {
+  // get hold of all tags and their corresponding input files using the literate config tool
+  const tagsMap = await getTagsMapFromConfig({ file: config });
+  if (tagsMap === null) {
+    return null;
+  }
+  const tags = Object.keys(tagsMap);
+
+  // filter the tags
+  if (utils.prOnly) {
```

Comment:
Checking only affected tags means we will be running the validator for contexts affected by the files changed in current PR and no other contexts, idea is to reduce noise as much as possible.